docs/configuration.md

```diff
@@ -326,4 +326,209 @@ deployment_config = DeploymentConfig(
        'docker_repository': 'your_repository',
        'gcp_project_id': '{project_id}',
        'dags_bucket': '{dags_bucket}'}})
-```
\ No newline at end of file
+```
+
+
+# Class-based configuration [experimental]
+
+Bigflow 1.3 introduces new class-based approach for keeping configurations.
+New API is optional and there is no need to migrate existing code from
+`biglfow.Configuration`. However it betters plays with typehinting, enables
+autocompletion in IDEs and gives more flexibility.
+
+Each configuration is declared as a subclass of `bigflow.konfig.Konfig`.
+Properties are declared as attributes, optionally with type hints.
+Classes may inherit from each other and share common properties.  Standard
+python inheritance rules are used to resolve overriden properties.
+Custom methods may be added to classes, however it is not recommended.
+
+```python
+import bigflow.konfig
+
+class CommonConf(bigflow.konfig.Konfig):
+    project: str                    # type hint
+    region: str = 'europe-west1'    # type hint + default value
+    machine_type = 'n2-standard-4'  # only value (type is `typing.Any`)
+    num_workers = 4                 # not only strings are allowed
+
+class DevConf(CommonConf):
+    project = "dev-project"
+
+class ProdConf(CommonConf):
+    project = "prod-project"
+    num_workers = 20
+```
+
+Config which corresponds to the current environment may be created by function `resolve_konfig`.
+It takes dictionary in form of {"environment" -> "config class"} and returns instance of the config.
```

Comment:
I think that you should mention that the configuration will be resolved based on the same environment variable mechanism, used in the "standard" bigflow.Configuration.

Reply:
done